### DISCUSSION TITLE
[V3] Strange modal behavior along with new navigation

### DISCUSSION THREAD

**Discussion by xp0nat** [livewire/livewire#6427] [1. Help]
Hello. i use new wire:navigate  to follow links.

I have a component that is responsible for opening and closing modals.

Everything is fine as long as you are on one page. But as soon as you go to another page, the modal window still opens and even a new post is added. But it stops working correctly and does not change the boolean value for closing the modal and opening a new modal on successful post addition.

Does anyone have any ideas, because I didn't find what I'm doing wrong in the documentation Livewire 3.

Thanks in advance for your time, I am attaching a video of my project on which I try it:

[Video](https://youtu.be/fmH7b3SlJO8)

**Comment by xp0nat** [comment]
Also, if I go from a certain page to another, where there is some kind of bootstrap js, then it no longer works until I refresh the page at this place.

**Reply by xp0nat** [reply]
After carefully reading the documentation, moving the bootstrap js to the < head > helped with the non-working js after navigation, but with the rest the question remains.

**Comment by danie-ramdhani** [comment]
close your modal after navigating:

```javascript
document.addEventListener('livewire:navigated', () => {
      closeModalFunction();
 });
```

**Reply by xp0nat** [reply]
You mean it should be like this?   
   
    public function createPostClose()
    {
        $this->createPost = false;
    }
    
    --------------------------------
    
    document.addEventListener('livewire:navigated', () => {
    createPostClose();
});


that way it didn't work, I saw it in the documentation but didn't quite understand how it should work, I don't have much experience.

**Reply by danie-ramdhani** [reply]
oh, you're using public property to toggle modal...
maybe somehting like this work:
```js
document.addEventListener('livewire:navigated', () => {
      $wire.$call('createPostClose');
 });
```

**Reply by xp0nat** [reply]
![image](https://github.com/livewire/livewire/assets/128713310/790d21e3-07c4-4efa-ae80-a336637e7bf4)


Right now it's causing these errors, the first is when navigating to another page, the second is when I open the modal on a new page. Anyway i think i need to dig in this direction, thanks.

**Reply by danie-ramdhani** [reply]
try this:
```javascript
document.addEventListener('livewire:navigated', () => {
      @this.call('createPostClose');
 });
```

**Reply by xp0nat** [reply]
This is similar to how it should work and doesn't cause errors, but it also doesn't solve the problem. By the way, when I go to a new page, I can open the modal, close it with a cross, then the modal doesn't open at all.

uncaught (in promise) undefined - the second time I press the button that is supposed to open the modal.

document.addEventListener('livewire:navigated', () => {
    Livewire.dispatch('createPostOpen');
    Livewire.dispatch('createPostClose');
});

**Reply by danie-ramdhani** [reply]
that because you don't have the similar property/function in your another component (don't do that).
i suggest you to learn the BASIC how to toggle something here:
https://alpinejs.dev/magics/dispatch

**Reply by xp0nat** [reply]
Thanks for your time and help. Today I updated the version and everything worked by itself. Apparently the effect of the beta version.
